src/udiskslinuxblock.c

```diff
@@ -2618,6 +2618,43 @@ handle_format_failure (GDBusMethodInvocation *invocation,
     g_clear_error (&error);
 }
 
+static gboolean
+add_blocksize (gchar        **command,
+               const gchar   *device,
+               GError       **error)
+{
+  gint fd = -1;
+  gchar *new_cmd = NULL;
+  gint blksize = 0;
+  gchar *size_str = NULL;
+
+  fd = open (device, O_RDONLY);
+  if (fd < 0)
+    {
+      g_set_error (error, UDISKS_ERROR, UDISKS_ERROR_FAILED,
+                   "Failed to open the device '%s' to get its block size", device);
+      return FALSE;
+    }
+
+  if (ioctl(fd, BLKSSZGET, &blksize) < 0)
```

Comment:
BLKSSZGET expects pointer to int, not to gint. As function ioctl() is variadic, there is no autoconversion you need to specify correct types.